crates/server/src/store/recursor.rs

```diff
@@ -290,11 +290,11 @@ impl RecursiveConfig {
 }
 
 fn recursion_limit_default() -> u8 {
-    12
+    24
```

Comment:
I think it would be good to isolate this change in a separate commit with a clear motivation in the commit message.

Reply:
The recursion limit change offsets the impact of the recursor logic change when handling very long CNAME chains, and thus splitting these into separate commits will require changing more tests (in some cases, once in each commit). I'm particularly having a hard time with the end-to-end test `recursor::cname::scenarios::multi_level_cname_tests`. Would you be alright with one commit if I add some more explanation to the commit message?

Reply:
Yeah, that sounds okay... flipflopping in test code doesn't sound appealing.